src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -98,6 +98,25 @@ internal static Type UnwrapTaskType(Type type)
             return null;
         }
 
+        internal static TAttribute GetResolvedAttribute<TAttribute>(ParameterInfo parameter) where TAttribute : Attribute
+        {
+            var attribute = parameter.GetCustomAttribute<TAttribute>();
+
+            var storageAttribute = attribute as StorageAttribute;
```

Comment:
>as StorageAttribute [](start = 45, length = 19)

This sort of special casing was what I wanted to avoid. (I think this is the crux of the difference between the 2 PRs). 
This is hard-coding the checks that I had previously generalized. In this case, it degenerates to the same runtime calls as the generalized algorithm. 
We'd need to duplicate these checks for ServiceBus.